T/tectonic/build_tarballs.jl

```diff
@@ -0,0 +1,50 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "tectonic"
+version = v"0.1.15"
+
+# Collection of sources required to build tar
+sources = [
+    ArchiveSource(
+        "https://github.com/tectonic-typesetting/tectonic/archive/tectonic@$(version).tar.gz",
+        "0e55188eafc1b58f3660a303fcdd6adc071051b9eb728119837fbeed2309914f"
+    )
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/tectonic-*/
+cargo build --release
+if [[ ${target} == *mingw* ]]; then
+    cp target/${rust_target}/release/tectonic.exe $bindir/
+else
+    cp target/${rust_target}/release/tectonic $bindir/
+fi
```

Comment:
[We have you covered](https://juliapackaging.github.io/BinaryBuilder.jl/dev/build_tips/#Automatic-environment-variables):
```suggestion
cp target/${rust_target}/release/tectonic${exeext} ${bindir}/
```